### PR TITLE
[Improve]Separate the record and schema-change in JsonDebeziumSchemaSerializer

### PR DESCRIPTION
**Pull request by DongLiang-0**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:
The original `JsonDebeziumSchemaSerializer` contains two serialization logics: record and schema change, which makes the entire `JsonDebeziumSchemaSerializer` code logic particularly confusing and complex, and the maintenance cost is high.

Now record and schema change are separated into `JsonDebeziumRecord` and `JsonDebeziumSchemaChange`. 
There are two versions of schema implementation in `JsonDebeziumSchemaChange`. One is the old version(`JsonDebeziumSchemaChangeImpl`), which has limited support for schema change in actual business; the other is a new version(`JsonDebeziumSchemaChangeImplV2`), which supports newer versions and can support multi-column changes and field renaming.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by JNSimba** [comment]
It seems that many JsonDebezium related classes have been added. Would it be better to create a separate package below?

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/JsonDebeziumSchemaChange.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import com.fasterxml.jackson.databind.node.NullNode;
+import org.apache.doris.flink.cfg.DorisOptions;
+import org.apache.doris.flink.sink.schema.SchemaChangeManager;
+import org.apache.doris.flink.tools.cdc.SourceSchema;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public abstract class JsonDebeziumSchemaChange implements Serializable {
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/JsonDebeziumSchemaChange.java:37]: we can add a comment at the beginning of the class to describe the function of the class

Comment (DongLiang-0) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/JsonDebeziumSchemaChange.java]: Sounds like a good suggestion, I will do.

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/jsondebezium/JsonDebeziumDataChange.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    public DorisRecord serialize(String record, JsonNode recordRoot, String op) throws IOException {
+        // Filter out table records that are not in tableMapping
+        String cdcTableIdentifier = getCdcTableIdentifier(recordRoot);
+        String dorisTableIdentifier = getDorisTableIdentifier(cdcTableIdentifier);
+        if (StringUtils.isNullOrWhitespaceOnly(dorisTableIdentifier)) {
+            LOG.warn(
+                    "filter table {}, because it is not listened, record detail is {}",
+                    cdcTableIdentifier,
+                    record);
+            return null;
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/jsondebezium/JsonDebeziumDataChange.java:83]: When adding tables dynamically, will the `tableMapping` here not change? So the data of the newly added table cannot be captured?

**Comment by JNSimba** [comment]
Please also rebase the master to run the latest workflow. Thanks

**Comment by DongLiang-0** [comment]
> Please also rebase the master to run the latest workflow. Thanks

Thanks for your suggestion, just done.

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/jsondebezium/JsonDebeziumDataChange.java (hunk 2 above)
Comment (DongLiang-0) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/jsondebezium/JsonDebeziumDataChange.java:78]: Thanks for your suggestion, it has been fixed.

**Comment by JNSimba** [comment]
Thanks Dongliang, can you add ITCase for this? like this https://github.com/apache/doris-flink-connector/blob/master/flink-doris-connector/src/test/java/org/apache/doris/flink/tools/cdc/MySQLDorisE2ECase.java

**Comment by DongLiang-0** [comment]
> Thanks Dongliang, can you add ITCase to the function of automatically adding tables? like this https://github.com/apache/doris-flink-connector/blob/master/flink-doris-connector/src/test/java/org/apache/doris/flink/tools/cdc/MySQLDorisE2ECase.java

Thanks for your suggestion, in the last commit, I have added.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**